providers/demiourgos.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { LockedEgldProvider } from '@libs/common';
+import { BigNumber } from 'bignumber.js';
+import axios from 'axios';
+
+@Injectable()
+export class DemiourgosProvider extends LockedEgldProvider {
+  private readonly tokenIdentifier = 'VEGLD-2b9319';
+  private readonly contracts = [
+    'erd1qqqqqqqqqqqqqpgqawus4zu5w2frmhh9rscjqnu9x6msfjya2d2sfw7tsn',
+  ];
+
+  init(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  // eslint-disable-next-line require-await
+  async getLockedEgldContracts(): Promise<string[]> {
+    return Promise.resolve(this.contracts);
+  }
+
+  async getAddressLockedEgld(address: string): Promise<{ lockedEgld: string }> {
+    const tokenBalance = await this.baseProvider.getTokenBalance(
+      address,
+      this.tokenIdentifier,
+    );
+
+    const response = await axios.get(
```

Comment:
2 things regarding this:
1. the price fetch can be done on the `init` function so it will be only called once and there will be the same ratio for all addresses
2. if possible, can we determine the ratio by performing gateway VM Queries? In the future, we'd like to add support for historical operations and this can only be done via VM Queries that can receive time parameters